From e3fc640d7a7fd99730eb7d28111d6eb830d39906 Mon Sep 17 00:00:00 2001
From: mydongistiny <jaysonedson@gmail.com>
Date: Sat, 21 Apr 2018 16:45:02 -0700
Subject: [PATCH 1/2] build: allow use of host flex binary

Arch Linux is using a newer glibc now so that causes build errors. Allow use of host flex to solve.

flex-2.5.39: loadlocale.c:130: _nl_intern_locale_data: Assertion `cnt < (sizeof (_nl_value_type_LC_TIME) / sizeof (_nl_value_type_LC_TIME[0]))' failed.

Change-Id: Id1154cfe9d7798570df7a81e6ae079b5f80d3f68
Signed-off-by: mydongistiny <jaysonedson@gmail.com>
---
 core/config.mk | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/core/config.mk b/core/config.mk
index 884495305a..a95aadb3a4 100644
--- a/core/config.mk
+++ b/core/config.mk
@@ -627,8 +627,11 @@ ZIPTIME := $(prebuilt_build_tools_bin)/ziptime
 
 # ---------------------------------------------------------------
 # Generic tools.
-
-LEX := $(prebuilt_build_tools_bin_noasan)/flex
+ifeq ($(USE_HOST_LEX),true)
+  LEX := flex
+else
+  LEX := prebuilts/misc/$(BUILD_OS)-$(HOST_PREBUILT_ARCH)/flex/flex-2.5.39
+endif
 # The default PKGDATADIR built in the prebuilt bison is a relative path
 # prebuilts/build-tools/common/bison.
 # To run bison from elsewhere you need to set up enviromental variable
-- 
2.17.1

